commit cb4bd0ec8dcba856d1ebf8b3f72b79f669dad0f4
Author: Stephen Hemminger <stephen.hemminger@vyatta.com>
Date:   Wed Aug 25 09:04:55 2010 -0700

    Fix GRED options clearing
    
    Bug reported where priorities of GRED DP's are ignored.
    The option parsing sets opt then memset was clearing these
    values.

diff --git a/tc/q_gred.c b/tc/q_gred.c
index df4aa3d..59651d3 100644
--- a/tc/q_gred.c
+++ b/tc/q_gred.c
@@ -51,7 +51,7 @@ static int init_gred(struct qdisc_util *qu, int argc, char **argv,
 {
 
 	struct rtattr *tail;
-	struct tc_gred_sopt opt;
+	struct tc_gred_sopt opt = { 0 };
 	int dps = 0;
 	int def_dp = -1;
 
@@ -83,7 +83,7 @@ static int init_gred(struct qdisc_util *qu, int argc, char **argv,
 				return -1;
 			}
 		} else if (strcmp(*argv, "grio") == 0) {
-			opt.grio=1;
+			opt.grio = 1;
 		} else if (strcmp(*argv, "help") == 0) {
 			explain();
 			return -1;
@@ -100,7 +100,6 @@ static int init_gred(struct qdisc_util *qu, int argc, char **argv,
 		return -1;
 	}
 
-	memset(&opt, 0, sizeof(struct tc_gred_sopt));
 	opt.DPs = dps;
 	opt.def_DP = def_dp;
 
